### PR TITLE
[POC] how to not be dependent on discovery

### PR DESCRIPTION
**Pull request by Nyholm**
This is just a proof of concept how you could do to not be dependent on discovery. This will make the dependency to puli or puli bundle optional

### CODE REVIEW

**Comment by dbu** [comment]
oh, that makes sense to me!

File: DependencyInjection/HttplugExtension.php [hunk 1]
```diff
@@ -207,4 +216,25 @@ private function configureAuthentication(ContainerBuilder $container, array $con
...
+
+    /**
+     * Verify that Puli is installed
+     *
+     * @param ContainerBuilder $container
+     * @throws \Exception
+     */
+    private function verifyDiscoveryInstalled(ContainerBuilder $container)
+    {
+        $enabledBundles = $container->getParameter('kernel.bundles');
```
Comment (dbu) [inline DependencyInjection/HttplugExtension.php]: i guess we need to fake some things for the unit tests here. we can then also test the error situation.

**Comment by sagikazarmark** [comment]
For me this is the same as the interface packages with one more downside: this way you have to manually install the Puli composer plugin AND the symfony bundle as well.

While the composer plugin is not necessary if you want to use puli, you can manually configure any kind of client, the symfony bundle is ALWAYS necessary in order to use Puli.

**Comment by sagikazarmark** [comment]
Hm, it seems I was wrong. The symfony-bundle actually requires the composer plugin. In this case, it's enough to suggest and install the symfony bundle.

File: DependencyInjection/HttplugExtension.php (hunk 1 above)
Comment (sagikazarmark) [inline DependencyInjection/HttplugExtension.php]: I don't think we need to check against the bundle, we can check against the service as well. I think that is easier to mock.

**Pull request closed without merging**